packages/vite/src/node/optimizer/scan.ts

```diff
@@ -242,7 +242,7 @@ function globEntries(pattern: string | string[], config: ResolvedConfig) {
   return glob(pattern, {
     cwd: config.root,
     ignore: [
-      '**/node_modules/**',
+      glob.convertPathToPattern(config.root) + '/**/node_modules/**',
```

Comment:
This shouldn't be needed. Could you explain why this is solving the issue? We are doing the glob with `cwd` at `config.root`, so if passing absolute paths to ignore is supported, it should give the same result as the one we already have here. Or there is a bug in `glob` or maybe passing this new pattern has the effect of invalidating the ignore for all users, so your app starts to work.

Reply:
Sure. Let me explain:

Let's say the full absolute path of `index.html` is: `/<my-workspace>/node_modules/@bitdev/react.examples.button/index.html`

Then the original `**/node_modules/**` will match this path and make it ignored by `fast-glob`.

However, `/<my-workspace>/node_modules/@bitdev/react.examples.button/**/node_modules/**` won't match it and it works.

Below is the log (before and after this PR):

Below:

```
input: {
  pattern: '**/*.html',
  cwd: '/Volumes/jinjiang-portable-disk/Developer/personal/reproductions/node_modules/.pnpm/@bitdev+react.examples.button@0.0.23_@testing-library+react@12.1.5_@types+react@18.2.48_react_5npp46vlhu2o2i5ubj23gk35sy/node_modules/@bitdev/react.examples.button',
  ignore: [
    '**/node_modules/**',
    '**/dist/**',
    '**/__tests__/**',
    '**/coverage/**'
  ],
  absolute: true,
  suppressErrors: true
}
result: { entries: [] }
```


After:

```
input: {
  pattern: '**/*.html',
  cwd: '/Volumes/jinjiang-portable-disk/Developer/personal/reproductions/node_modules/.pnpm/@bitdev+react.examples.button@0.0.23_@testing-library+react@12.1.5_@types+react@18.2.48_react_5npp46vlhu2o2i5ubj23gk35sy/node_modules/@bitdev/react.examples.button',
  ignore: [
    '/Volumes/jinjiang-portable-disk/Developer/personal/reproductions/node_modules/.pnpm/@bitdev+react.examples.button@0.0.23_@testing-library+react@12.1.5_@types+react@18.2.48_react_5npp46vlhu2o2i5ubj23gk35sy/node_modules/@bitdev/react.examples.button/**/node_modules/**',
    '**/dist/**',
    '**/__tests__/**',
    '**/coverage/**'
  ],
  absolute: true,
  suppressErrors: true
}
result: {
  entries: [
    '/Volumes/jinjiang-portable-disk/Developer/personal/reproductions/node_modules/.pnpm/@bitdev+react.examples.button@0.0.23_@testing-library+react@12.1.5_@types+react@18.2.48_react_5npp46vlhu2o2i5ubj23gk35sy/node_modules/@bitdev/react.examples.button/index.html'
  ]
}
```

Reply:
Ok, I see 👍🏼